social_protection/gql_mutations.py

```diff
@@ -178,12 +178,16 @@ def _mutate(cls, user, **data):
             data.pop('client_mutation_label')
 
         service = BenefitPlanService(user)
-
+        res = {'success': False, 'message': 'No IDs to delete', 'details': ''}
         ids = data.get('ids')
         if ids:
             with transaction.atomic():
-                for id in ids:
-                    service.delete({'id': id})
+                for obj_id in ids:
+                    res = service.delete({'id': obj_id})
+                    if not res['success']:
+                        transaction.rollback()
+                        break
+        return res if not res['success'] else None
```

Comment:
Thanks for this improvement. In terms of code I think we could reduce the indentation with something like: 
```python
    ids = data.get('ids')
    if not ids:
        return {'success': False, 'message': 'No IDs to delete', 'details': ''}

    service = BenefitPlanService(user)
    with transaction.atomic():
        for obj_id in ids:
            res = service.delete({'id': obj_id})
            if not res['success']:
                transaction.rollback()
                return res

    return None
```
let me know your thoughts.

Reply:
Same would apply for remaining mutations that were changed.

Reply:
I like this version more